src/storage/FileResourceStore.ts

```diff
@@ -0,0 +1,427 @@
+  };
+
+  /**
+   * @param baseRequestURI - Will be stripped of all incoming URIs and added to all outgoing ones to find the relative
+   * path.
+   * @param rootFilepath - Root filepath in which the resources and containers will be saved as files and directories.
+   * @param interactionController - Instance of InteractionController to use.
+   */
+  public constructor(baseRequestURI: string, rootFilepath: string, interactionController: InteractionController) {
+    this.baseRequestURI = trimTrailingSlashes(baseRequestURI);
+    this.rootFilepath = trimTrailingSlashes(rootFilepath);
+    this.interactionController = interactionController;
+  }
+
+  /**
+   * Store the incoming data as a file under a file path corresponding to `container.path`,
+   * where slashes correspond to subdirectories.
+   * @param container - The identifier to store the new data under.
+   * @param representation - Data to store. Only File streams are supported.
+   *
+   * @returns The newly generated identifier.
+   */
+  public async addResource(container: ResourceIdentifier, representation: Representation): Promise<ResourceIdentifier> {
+    if (representation.dataType !== DATA_TYPE_BINARY) {
+      throw new UnsupportedMediaTypeHttpError('FileResourceStore only supports binary representations.');
+    }
+    const path = this.parseIdentifier(container);
+    const { slug, raw } = representation.metadata;
+    const linkTypes = representation.metadata.linkRel?.type;
+    let metadata;
+    if (raw.length > 0) {
+      metadata = streamifyArray(raw).pipe(new StreamWriter({ format: 'text/turtle' }));
+    }
+
+    const isContainer = this.interactionController.isContainer(slug, linkTypes);
+    const newIdentifier = this.interactionController.generateIdentifier(isContainer, slug);
+    if (!isContainer) {
+      // Create a file for the resource with as filepath the parent container.
+      return this.createFile(path, newIdentifier, representation.data, path.endsWith('/'), metadata);
+    }
+
+    // Create a new container as subdirectory of the parent container.
+    return this.createContainer(path, newIdentifier, path.endsWith('/'), metadata);
+  }
+
+  /**
+   * Deletes the given resource.
+   * @param identifier - Identifier of resource to delete.
+   */
+  public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
+    let path = this.parseIdentifier(identifier);
+    if (path === '' || ensureTrailingSlash(path) === '/') {
+      throw new MethodNotAllowedHttpError('Cannot delete root container.');
+    }
+    path = joinPath(this.rootFilepath, path);
+    let stats;
+    try {
+      stats = await fsPromises.lstat(path);
+    } catch (error) {
+      throw new NotFoundHttpError();
+    }
+    if (stats.isFile()) {
+      await fsPromises.unlink(path);
+      try {
+        await fsPromises.unlink(`${path}.metadata`);
+      } catch (_) {
+        // It's ok if there was no metadata file.
```

Comment:
@joachimvh Do we have a logging mechanism in the project? (If not, let's create an issue now!) I'd think about adding an `info` message here.

Reply:
No logging yet

Reply:
https://github.com/solid/community-server/issues/54